### ISSUE TITLE
[SP5]

### ISSUE THREAD

**Issue by ArPhil** [linux-surface/linux-surface#933]
Hi,

First of all: Thanks a lot for all the effort and work that was put into the linux-surface project. I highly appreciate it and enjoy using my Surface Pro with Linux very much.

I have a input regarding touch inputs similar to the following ones:

- https://github.com/linux-surface/linux-surface/issues/781
- https://github.com/linux-surface/linux-surface/issues/925
- https://github.com/linux-surface/linux-surface/issues/389

Not sure with with (kernel) update it happened, but since a while I noticed much worse touch input recognition than I had earlier. I suspect (as mentioned in the other issues above), that it is connected to thumb and / or palm detection. The reasons I assume this, are:

1. When using any finger (but thumb) and pressing gently on the touchscreen, everything works fine, even touchscreen gestures like accessing the application menu on ubuntu.
2. When I use one of my thumbs, touch recognition is much worse and mostly no touch decetion happes. I have to press VERY (!) gently with my thumb in order to get any response on the screen (probably because then my thumb is detected as "regular" (non thumb) finger.
3. When touching with my palm, nothing happens, so palm detection seems to work fine.

I tried to follow the hints in the issues mentioned above and tried to change the config in `/etc/ipts.conf`. I tried to use values given in https://github.com/linux-surface/iptsd/blob/master/etc/ipts.conf. Especially I tried to change `FingerSize` and `ThumbSize` to higher values. But it seems that none of theses changes makes it better. It even seems that the changes are ignored entirely.

**How can I change the settings for finger / thumb / palm detection for my SP5? Is there any documentation what (config) file to change and how make the ipts driver use my values?** 

The ipts process is running as `root`: `/usr/bin/iptsd`

I`d appreciate any feedback! Thanks!

### Environment

Linux surfux 5.19.8-surface #1 SMP PREEMPT_DYNAMIC Wed Sep 14 22:12:13 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

- Hardware model: Microsoft Surface Pro 5, 8 GB Ram, 256 GB SSD, LTE / 4G 
- Kernel version: Linux surfux 5.19.8-surface #1 SMP PREEMPT_DYNAMIC Wed Sep 14 22:12:13 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
- Distribution: Ubuntu 22.04.1 LTS

<details><summary>`dmesg` output</summary>

```
-

```

</details>

**Comment by zccrs** [comment]
I can't reproduction this issue on 6.2.2-surface

**Comment by tommeer** [comment]
Hi, I would like to add another data point for a Surface Pro 5 (2017).

Device:
- Microsoft Surface Pro 5 / 2017
- Intel Core i5-7300U
- 8 GB RAM
- Intel HD Graphics 620
- 256 GB SSD

Software:
- Ubuntu
- linux-surface kernel: 6.18.7-surface-1

Current status:
- Wi-Fi works
- Bluetooth is detected
- Audio works
- Type Cover keyboard works
- Type Cover touchpad works
- Volume buttons work
- Accelerometer is detected and monitor-sensor reports orientation changes correctly
- Camera devices appear, but camera is not working yet
- Touchscreen is detected and partially working, but finger input is still not working well enough for normal tablet-style use

Touch-specific observations:
- Touch is no longer completely dead after installing the linux-surface kernel
- The following devices are present:
  - IPTS 045E:001F Touchscreen
  - IPTSD Virtual Touchscreen 045E:001F
  - IPTSD Virtual Stylus 045E:001F
- The iptsd service is active and running:
  - iptsd@dev-hidraw2.service
- iptsd connects successfully to the device and reports:
  - Connected to device 045E:001F
  - Running in Touchscreen mode
- Raw input seems to be processed correctly with iptsd-show

Remaining problem:
- Finger touch still feels unreliable/inaccurate
- One-finger scrolling is not smooth
- Text selection with a finger is difficult and not reliable
- General touch interactions are only partially usable
- Other interactions seem better than scrolling/selecting text, so the stack is not completely broken

Relevant logs:
- Kernel: 6.18.7-surface-1
- iptsd loads:
  - /usr/share/iptsd/surface-pro-6.conf
  - /etc/iptsd.conf

Additional note:
- monitor-sensor reports accelerometer changes correctly (for example normal <-> left-up), so sensor detection itself seems fine

So in short:
the linux-surface kernel and iptsd clearly improve the situation and make touch available, but finger input quality on the Surface Pro 5 is still not good enough, especially for one-finger scrolling and text selection.

If needed, I can provide full command output for:
- uname -r
- libinput list-devices
- systemctl status iptsd@dev-hidraw2.service
- journalctl logs
- monitor-sensor
